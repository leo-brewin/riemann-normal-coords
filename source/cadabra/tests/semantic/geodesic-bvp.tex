\def\Date{19 Jan 2024}
% \def\FileID{file:}

\documentclass[12pt]{cdblatex}

\begin{document}

\section*{\jobname}

\CdbSetup{action=hide}

\begin{cadabra}
   import shared

   import cdblib

   term00A = cdblib.get ('check000','expected/geodesic-bvp.json')
   term01A = cdblib.get ('check001','expected/geodesic-bvp.json')
   term02A = cdblib.get ('check002','expected/geodesic-bvp.json')
   term03A = cdblib.get ('check003','expected/geodesic-bvp.json')
   term04A = cdblib.get ('check004','expected/geodesic-bvp.json')
   term05A = cdblib.get ('check005','expected/geodesic-bvp.json')
   term06A = cdblib.get ('check005','expected/geodesic-bvp.json')
   term07A = cdblib.get ('check005','expected/geodesic-bvp.json')
   term08A = cdblib.get ('check005','expected/geodesic-bvp.json')
   term09A = cdblib.get ('check005','expected/geodesic-bvp.json')
   term10A = cdblib.get ('check005','expected/geodesic-bvp.json')
   term00B = cdblib.get ('check000','output/geodesic-bvp.json')
   term01B = cdblib.get ('check001','output/geodesic-bvp.json')
   term02B = cdblib.get ('check002','output/geodesic-bvp.json')
   term03B = cdblib.get ('check003','output/geodesic-bvp.json')
   term04B = cdblib.get ('check004','output/geodesic-bvp.json')
   term05B = cdblib.get ('check005','output/geodesic-bvp.json')
   term06B = cdblib.get ('check005','output/geodesic-bvp.json')
   term07B = cdblib.get ('check005','output/geodesic-bvp.json')
   term08B = cdblib.get ('check005','output/geodesic-bvp.json')
   term09B = cdblib.get ('check005','output/geodesic-bvp.json')
   term10B = cdblib.get ('check005','output/geodesic-bvp.json')

   diff000 = shared.check (term00A,term00B)   # cdb (diff000,diff000)
   diff001 = shared.check (term01A,term01B)   # cdb (diff001,diff001)
   diff002 = shared.check (term02A,term02B)   # cdb (diff002,diff002)
   diff003 = shared.check (term03A,term03B)   # cdb (diff003,diff003)
   diff004 = shared.check (term04A,term04B)   # cdb (diff004,diff004)
   diff005 = shared.check (term05A,term05B)   # cdb (diff005,diff005)
   diff006 = shared.check (term06A,term06B)   # cdb (diff006,diff006)
   diff007 = shared.check (term07A,term07B)   # cdb (diff007,diff007)
   diff008 = shared.check (term08A,term08B)   # cdb (diff008,diff008)
   diff009 = shared.check (term09A,term09B)   # cdb (diff009,diff009)
   diff010 = shared.check (term10A,term10B)   # cdb (diff010,diff010)

\end{cadabra}

\begin{dgroup*}
   \Dmath*{ \cdb*{diff000} }
   \Dmath*{ \cdb*{diff001} }
   \Dmath*{ \cdb*{diff002} }
   \Dmath*{ \cdb*{diff003} }
   \Dmath*{ \cdb*{diff004} }
   \Dmath*{ \cdb*{diff005} }
   \Dmath*{ \cdb*{diff006} }
   \Dmath*{ \cdb*{diff007} }
   \Dmath*{ \cdb*{diff008} }
   \Dmath*{ \cdb*{diff009} }
   \Dmath*{ \cdb*{diff010} }
\end{dgroup*}

\end{document}
