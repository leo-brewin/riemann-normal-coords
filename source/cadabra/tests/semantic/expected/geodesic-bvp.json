{
  "check000": "x^{a} + \\lam Dx^{a} -  1/24 (\\lam-(\\lam)**2) Dx^{b} Dx^{c} (8x^{d} g^{a e} R_{d b c e} + x^{d} x^{e} (2g^{a f} \\nabla_{b}(R_{d c e f}) + 4g^{a f} \\nabla_{d}(R_{e b c f})-g^{a f} \\nabla_{f}(R_{d b e c}))) -  1/12 (\\lam-(\\lam)**3) x^{b} Dx^{c} Dx^{d} Dx^{e} g^{a f} \\nabla_{c}(R_{b d e f})",
  "check001": "x^{b} Dx^{c} Dx^{d} g^{a e} R_{b c d e}",
  "check002": "x^{b} x^{c} Dx^{d} Dx^{e} (2g^{a f} \\nabla_{d}(R_{b e c f}) + 4g^{a f} \\nabla_{b}(R_{c d e f})-g^{a f} \\nabla_{f}(R_{b d c e}))",
  "check003": "x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} (80g^{a g} g^{h i} R_{b d e h} R_{c f g i}-80g^{a g} g^{h i} R_{b d e h} R_{c i f g}) + x^{b} x^{c} x^{d} Dx^{e} Dx^{f} (64g^{a g} g^{h i} R_{b e f h} R_{c g d i}-32g^{a g} g^{h i} R_{b e c h} R_{d i f g}-16g^{a g} g^{h i} R_{b e c h} R_{d g f i} + 18g^{a g} \\nabla_{e b}(R_{c f d g}) + 18g^{a g} \\nabla_{b e}(R_{c f d g}) + 36g^{a g} \\nabla_{b c}(R_{d e f g}) + 16g^{a g} g^{h i} R_{b e c h} R_{d f g i}-9g^{a g} \\nabla_{g b}(R_{c e d f})-9g^{a g} \\nabla_{b g}(R_{c e d f}))",
  "check004": "x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} (10g^{a h} g^{i j} R_{b e h i} \\nabla_{f}(R_{c g d j}) + 20g^{a h} g^{i j} R_{b e h i} \\nabla_{c}(R_{d f g j})-5g^{a h} g^{i j} R_{b e h i} \\nabla_{j}(R_{c f d g})-10g^{a h} g^{i j} R_{b i e h} \\nabla_{f}(R_{c g d j})-20g^{a h} g^{i j} R_{b i e h} \\nabla_{c}(R_{d f g j}) + 5g^{a h} g^{i j} R_{b i e h} \\nabla_{j}(R_{c f d g})-10g^{a h} g^{i j} R_{b e f i} \\nabla_{g}(R_{c h d j})-10g^{a h} g^{i j} R_{b e f i} \\nabla_{j}(R_{c g d h}) + 20g^{a h} g^{i j} R_{b e f i} \\nabla_{c}(R_{d g h j})-20g^{a h} g^{i j} R_{b e f i} \\nabla_{c}(R_{d j g h}) + 10g^{a h} g^{i j} R_{b e f i} \\nabla_{h}(R_{c g d j})) + x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} (10g^{a h} g^{i j} R_{b d h i} \\nabla_{e}(R_{c f g j})-10g^{a h} g^{i j} R_{b i d h} \\nabla_{e}(R_{c f g j})) + x^{b} x^{c} x^{d} x^{e} Dx^{f} Dx^{g} (16g^{a h} g^{i j} R_{b f g i} \\nabla_{c}(R_{d h e j}) + 6g^{a h} g^{i j} R_{b h c i} \\nabla_{f}(R_{d g e j}) + 16g^{a h} g^{i j} R_{b h c i} \\nabla_{d}(R_{e f g j})-5g^{a h} g^{i j} R_{b h c i} \\nabla_{j}(R_{d f e g})-8g^{a h} g^{i j} R_{b i f h} \\nabla_{c}(R_{d g e j})-4g^{a h} g^{i j} R_{b h f i} \\nabla_{c}(R_{d g e j})-4g^{a h} g^{i j} R_{b f c i} \\nabla_{g}(R_{d h e j})-8g^{a h} g^{i j} R_{b f c i} \\nabla_{d}(R_{e j g h})-4g^{a h} g^{i j} R_{b f c i} \\nabla_{d}(R_{e h g j}) + 2g^{a h} \\nabla_{f b c}(R_{d g e h}) + 2g^{a h} \\nabla_{b f c}(R_{d g e h}) + 2g^{a h} \\nabla_{b c f}(R_{d g e h}) + 4g^{a h} \\nabla_{b c d}(R_{e f g h}) + 4g^{a h} g^{i j} R_{b f h i} \\nabla_{c}(R_{d g e j}) + 4g^{a h} g^{i j} R_{b f c i} \\nabla_{h}(R_{d g e j}) + 4g^{a h} g^{i j} R_{b f c i} \\nabla_{d}(R_{e g h j})-g^{a h} \\nabla_{h b c}(R_{d f e g})-g^{a h} \\nabla_{b h c}(R_{d f e g})-g^{a h} \\nabla_{b c h}(R_{d f e g}))",
  "check005": "x^{b} Dx^{c} Dx^{d} Dx^{e} g^{a f} \\nabla_{c}(R_{b d e f})",
  "check006": "x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} (64g^{a g} g^{h i} R_{b d e h} R_{c i f g} + 16g^{a g} g^{h i} R_{b d e h} R_{c g f i}-16g^{a g} g^{h i} R_{b d c h} R_{e g f i} + 12g^{a g} \\nabla_{d e}(R_{b f c g}) + 18g^{a g} \\nabla_{d b}(R_{c e f g}) + 18g^{a g} \\nabla_{b d}(R_{c e f g})-48g^{a g} g^{h i} R_{b d e h} R_{c f g i}-3g^{a g} \\nabla_{g d}(R_{b e c f})-3g^{a g} \\nabla_{d g}(R_{b e c f}))",
  "check007": "x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} (30g^{a h} g^{i j} R_{b d e i} \\nabla_{f}(R_{c g h j})-30g^{a h} g^{i j} R_{b d e i} \\nabla_{f}(R_{c j g h})-30g^{a h} g^{i j} R_{b d e i} \\nabla_{j}(R_{c f g h})) + x^{b} x^{c} x^{d} Dx^{e} Dx^{f} Dx^{g} (32g^{a h} g^{i j} R_{b e f i} \\nabla_{g}(R_{c h d j}) + 48g^{a h} g^{i j} R_{b e f i} \\nabla_{c}(R_{d j g h}) + 12g^{a h} g^{i j} R_{b e f i} \\nabla_{c}(R_{d h g j}) + 18g^{a h} g^{i j} R_{b i e h} \\nabla_{f}(R_{c g d j}) + 2g^{a h} g^{i j} R_{b h e i} \\nabla_{f}(R_{c g d j}) + 22g^{a h} g^{i j} R_{b h c i} \\nabla_{e}(R_{d f g j}) + 48g^{a h} g^{i j} R_{b i e h} \\nabla_{c}(R_{d f g j}) + 12g^{a h} g^{i j} R_{b h e i} \\nabla_{c}(R_{d f g j})-15g^{a h} g^{i j} R_{b i e h} \\nabla_{j}(R_{c f d g})-5g^{a h} g^{i j} R_{b h e i} \\nabla_{j}(R_{c f d g})-12g^{a h} g^{i j} R_{e h f i} \\nabla_{b}(R_{c g d j})-12g^{a h} g^{i j} R_{b e c i} \\nabla_{f}(R_{d j g h})-8g^{a h} g^{i j} R_{b e c i} \\nabla_{f}(R_{d h g j})-12g^{a h} g^{i j} R_{b e c i} \\nabla_{d}(R_{f h g j}) + 4g^{a h} \\nabla_{e f b}(R_{c g d h}) + 4g^{a h} \\nabla_{e b f}(R_{c g d h}) + 6g^{a h} \\nabla_{e b c}(R_{d f g h}) + 4g^{a h} \\nabla_{b e f}(R_{c g d h}) + 6g^{a h} \\nabla_{b e c}(R_{d f g h}) + 6g^{a h} \\nabla_{b c e}(R_{d f g h})-16g^{a h} g^{i j} R_{b e h i} \\nabla_{f}(R_{c g d j})-36g^{a h} g^{i j} R_{b e h i} \\nabla_{c}(R_{d f g j})-16g^{a h} g^{i j} R_{b e f i} \\nabla_{h}(R_{c g d j}) + 4g^{a h} g^{i j} R_{b e c i} \\nabla_{h}(R_{d f g j})-36g^{a h} g^{i j} R_{b e f i} \\nabla_{c}(R_{d g h j}) + 4g^{a h} g^{i j} R_{b e c i} \\nabla_{f}(R_{d g h j})-g^{a h} \\nabla_{h e b}(R_{c f d g})-g^{a h} \\nabla_{h b e}(R_{c f d g})-g^{a h} \\nabla_{e h b}(R_{c f d g})-g^{a h} \\nabla_{b h e}(R_{c f d g})-g^{a h} \\nabla_{e b h}(R_{c f d g})-g^{a h} \\nabla_{b e h}(R_{c f d g}) + 20g^{a h} g^{i j} R_{b e f i} \\nabla_{j}(R_{c g d h}) + 10g^{a h} g^{i j} R_{b e h i} \\nabla_{j}(R_{c f d g}))",
  "check008": "x^{b} Dx^{c} Dx^{d} Dx^{e} Dx^{f} (4g^{a g} g^{h i} R_{b c d h} R_{e g f i} + 3g^{a g} \\nabla_{c d}(R_{b e f g}))",
  "check009": "x^{b} x^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} (64g^{a h} g^{i j} R_{b d e i} \\nabla_{f}(R_{c j g h}) + 18g^{a h} g^{i j} R_{b d e i} \\nabla_{f}(R_{c h g j}) + 24g^{a h} g^{i j} R_{b d e i} \\nabla_{c}(R_{f h g j}) + 4g^{a h} g^{i j} R_{d h e i} \\nabla_{f}(R_{b g c j}) + 44g^{a h} g^{i j} R_{b i d h} \\nabla_{e}(R_{c f g j}) + 18g^{a h} g^{i j} R_{b h d i} \\nabla_{e}(R_{c f g j}) + 24g^{a h} g^{i j} R_{d h e i} \\nabla_{b}(R_{c f g j})-10g^{a h} g^{i j} R_{d h e i} \\nabla_{j}(R_{b f c g})-16g^{a h} g^{i j} R_{b d c i} \\nabla_{e}(R_{f h g j}) + 6g^{a h} \\nabla_{d e f}(R_{b g c h}) + 8g^{a h} \\nabla_{d e b}(R_{c f g h}) + 8g^{a h} \\nabla_{d b e}(R_{c f g h}) + 8g^{a h} \\nabla_{b d e}(R_{c f g h})-26g^{a h} g^{i j} R_{b d h i} \\nabla_{e}(R_{c f g j})-6g^{a h} g^{i j} R_{b d e i} \\nabla_{h}(R_{c f g j})-46g^{a h} g^{i j} R_{b d e i} \\nabla_{f}(R_{c g h j})-g^{a h} \\nabla_{h d e}(R_{b f c g})-g^{a h} \\nabla_{d h e}(R_{b f c g})-g^{a h} \\nabla_{d e h}(R_{b f c g}) + 40g^{a h} g^{i j} R_{b d e i} \\nabla_{j}(R_{c f g h}))",
  "check010": "x^{b} Dx^{c} Dx^{d} Dx^{e} Dx^{f} Dx^{g} (3g^{a h} g^{i j} R_{b c d i} \\nabla_{e}(R_{f h g j}) + 3g^{a h} g^{i j} R_{c h d i} \\nabla_{e}(R_{b f g j}) + g^{a h} \\nabla_{c d e}(R_{b f g h}))"
}
